src/meta-srv/src/handler.rs

```diff
@@ -489,50 +487,128 @@ impl HeartbeatHandlerGroupBuilder {
         } else {
             None
         };
 
-        // TODO(weny): Considers classifying handlers
-        // to make it easier for upper layers to customize handler groups.
-        let mut group = HeartbeatHandlerGroup::new(self.pushers);
-        group.add_handler(ResponseHeaderHandler);
+        self.add_handler(ResponseHeaderHandler);
         // `KeepLeaseHandler` should preferably be in front of `CheckLeaderHandler`,
         // because even if the current meta-server node is no longer the leader it can
         // still help the datanode to keep lease.
-        group.add_handler(DatanodeKeepLeaseHandler);
-        group.add_handler(FlownodeKeepLeaseHandler);
-        group.add_handler(CheckLeaderHandler);
-        group.add_handler(OnLeaderStartHandler);
-        group.add_handler(ExtractStatHandler);
-        group.add_handler(CollectDatanodeClusterInfoHandler);
-        group.add_handler(CollectFrontendClusterInfoHandler);
-        group.add_handler(CollectFlownodeClusterInfoHandler);
-        group.add_handler(MailboxHandler);
-        group.add_handler(self.region_lease_handler);
-        group.add_handler(FilterInactiveRegionStatsHandler);
-        if let Some(region_failure_handler) = self.region_failure_handler {
-            group.add_handler(region_failure_handler);
+        self.add_handler(DatanodeKeepLeaseHandler);
+        self.add_handler(FlownodeKeepLeaseHandler);
+        self.add_handler(CheckLeaderHandler);
+        self.add_handler(OnLeaderStartHandler);
+        self.add_handler(ExtractStatHandler);
+        self.add_handler(CollectDatanodeClusterInfoHandler);
+        self.add_handler(CollectFrontendClusterInfoHandler);
+        self.add_handler(CollectFlownodeClusterInfoHandler);
+        self.add_handler(MailboxHandler);
+        if let Some(region_lease_handler) = self.region_lease_handler.take() {
+            self.add_handler(region_lease_handler);
+        }
+        self.add_handler(FilterInactiveRegionStatsHandler);
+        if let Some(region_failure_handler) = self.region_failure_handler.take() {
+            self.add_handler(region_failure_handler);
         }
         if let Some(publish_heartbeat_handler) = publish_heartbeat_handler {
-            group.add_handler(publish_heartbeat_handler);
+            self.add_handler(publish_heartbeat_handler);
         }
-        group.add_handler(CollectStatsHandler::default());
+        self.add_handler(CollectStatsHandler::default());
 
-        group
+        self
+    }
+
+    /// Builds the group of heartbeat handlers.
+    pub fn build(self) -> HeartbeatHandlerGroup {
+        HeartbeatHandlerGroup {
+            handlers: self.handlers.into_iter().collect(),
+            pushers: self.pushers,
+        }
+    }
+
+    /// Adds the handler after the specified handler.
+    pub fn add_handler_after(
+        &mut self,
+        handler: impl HeartbeatHandler + 'static,
+        after: &'static str,
```

Comment:
```suggestion
        name: &'static str,
```

Let's place the name argument before the handler.